### DISCUSSION TITLE
insert into select报错 CAUSED BY: SQLException: GC overhead limit exceeded CAUSED BY: OutOfMemoryError: GC overhead limit exceeded

### DISCUSSION THREAD

**Discussion by 442496146** [apache/doris#16805] [A - General / Q&A] [answered]
I0215 17:37:02.674711 49364 olap_server.cpp:719] cooldown producer get tablet num: 0
I0215 17:37:04.918622 48861 daemon.cpp:217] physical memory 15.65 GB, process memory used 1.97 GB limit 12.52 GB, sys mem available 11.23 GB low water mark 1.57 GB, refresh interval memory growth 0 B
I0215 17:37:09.956616 48861 daemon.cpp:217] physical memory 15.65 GB, process memory used 1.97 GB limit 12.52 GB, sys mem available 11.23 GB low water mark 1.57 GB, refresh interval memory growth 0 B
I0215 17:37:10.827906 49446 tablet_manager.cpp:868] find expired transactions for 0 tablets
I0215 17:37:10.828029 49446 tablet_manager.cpp:906] success to build all report tablets info. tablet_count=60
I0215 17:37:10.829087 49446 task_worker_pool.cpp:1519] successfully report TABLET|host=192.168.129.101|port=9020
W0215 17:37:10.894472 48947 jni-util.cpp:193] org.apache.doris.udf.UdfRuntimeException: JDBC executor sql has error: 
	at org.apache.doris.udf.JdbcExecutor.read(JdbcExecutor.java:95)
Caused by: java.sql.SQLException: GC overhead limit exceeded
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129)
	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:953)
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeQuery(ClientPreparedStatement.java:1003)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeQuery(ProxyPreparedStatement.java:52)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeQuery(HikariProxyPreparedStatement.java)
	at org.apache.doris.udf.JdbcExecutor.read(JdbcExecutor.java:91)
Caused by: java.lang.OutOfMemoryError: GC overhead limit exceeded
W0215 17:37:10.895042 49070 fragment_mgr.cpp:391] report error status: Internal error: UdfRuntimeException: JDBC executor sql has error: 
CAUSED BY: SQLException: GC overhead limit exceeded
CAUSED BY: OutOfMemoryError: GC overhead limit exceeded to coordinator: TNetworkAddress(hostname=192.168.129.101, port=9020), query id: 54366bcc3d7a4915-a3cfb03428ce250e, instance id: 54366bcc3d7a4915-a3cfb03428ce250f
W0215 17:37:10.896625 49070 fragment_mgr.cpp:250] Got error while opening fragment 54366bcc3d7a4915-a3cfb03428ce250f, query id: 54366bcc3d7a4915-a3cfb03428ce250e: Internal error: UdfRuntimeException: JDBC executor sql has error: 
CAUSED BY: SQLException: GC overhead limit exceeded
CAUSED BY: OutOfMemoryError: GC overhead limit exceeded
W0215 17:37:10.908133 49070 tablet_sink.cpp:156] cancel node channel VNodeChannel[13280-11002], load_id=54366bcc3d7a4915-a3cfb03428ce250e, txn_id=2017, node=192.168.129.100:8060, error message: UdfRuntimeException: JDBC executor sql has error: 
CAUSED BY: SQLException: GC overhead limit exceeded
CAUSED BY: OutOfMemoryError: GC overhead limit exceeded
I0215 17:37:10.908202 49070 tablet_sink.cpp:1244] finished to close olap table sink. load_id=54366bcc3d7a4915-a3cfb03428ce250e, txn_id=2017, canceled all node channels due to error: UdfRuntimeException: JDBC executor sql has error: 
CAUSED BY: SQLException: GC overhead limit exceeded
CAUSED BY: OutOfMemoryError: GC overhead limit exceeded
I0215 17:37:10.908288 49539 load_channel_mgr.cpp:151] load channel has been cancelled: 54366bcc3d7a4915-a3cfb03428ce250e
I0215 17:37:10.908291 49539 load_channel.cpp:44] load channel removed. mem peak usage=0, info=label: LoadChannel#senderIp=192.168.129.100#loadID=54366bcc3d7a4915-a3cfb03428ce250e; consumption: 0; peak_consumption: 0; , load_id=54366bcc3d7a4915-a3cfb03428ce250e, is high priority=0, sender_ip=192.168.129.100, is_vec=1
I0215 17:37:10.908506 49070 plan_fragment_executor.cpp:699] Fragment 54366bcc3d7a4915-a3cfb03428ce250f:(Active: 1m44s, non-child: 0.00%)
   - FragmentCpuTime: 1.045ms
   - RowsProduced: 0
  BlockMgr:
     - BlockWritesOutstanding: 0
     - BlocksCreated: 0
     - BlocksRecycled: 0
     - BufferedPins: 0
     - BytesWritten: 0
     - MaxBlockSize: 8.00 MB
     - TotalBufferWaitTime: 0.000ns
     - TotalEncryptionTime: 0.000ns
     - TotalIntegrityCheckTime: 0.000ns
     - TotalReadBlockTime: 0.000ns
  OlapTableSink:(Active: 1.161ms, non-child: 0.00%)
     - CloseWaitTime: 0.000ns
     - ConvertBatchTime: 0.000ns
     - MaxAddBatchExecTime: 0.000ns
     - NonBlockingSendTime: 1m44s
       - NonBlockingSendWorkTime: 0.000ns
         - SerializeBatchTime: 0.000ns
     - NumberBatchAdded: 0
     - NumberNodeChannels: 0
     - OpenTime: 943.473us
     - RowsFiltered: 0
     - RowsRead: 0
     - RowsReturned: 0
     - SendDataTime: 0.000ns
       - WaitMemLimitTime: 0.000ns
     - TotalAddBatchExecTime: 0.000ns
     - ValidateDataTime: 0.000ns
  VNewJdbcScanNode(`test`.`t1`) (id=0):(Active: 1m44s, non-child: 100.00%)
     - AcuireRuntimeFilterTime: 907.000ns
     - GetNextTime: 1m44s
     - MaxScannerThreadNum: 1
     - NewlyCreateFreeBlocksNum: 0
     - NumScanners: 1
     - PeakMemoryUsage: 1.01 MB
     - PreAllocFreeBlocksNum: 17
     - ProjectionTime: 0.000ns
     - RowsRead: 0
     - RowsReturned: 0
     - RowsReturnedRate: 0
     - ScannerBatchWaitTime: 1m44s
     - ScannerCtxSchedCount: 2
     - ScannerSchedCount: 1
     - ScannerWorkerWaitTime: 121.753us
     - TotalReadThroughput: 0
    VScanner:
       - ScannerConvertBlockTime: 0.000ns
       - ScannerCpuTime: 1s340ms
       - ScannerFilterTime: 0.000ns
       - ScannerGetBlockTime: 0.000ns
       - ScannerPrefilterTime: 0.000ns
I0215 17:37:10.908524 49070 plan_fragment_executor.cpp:701] Close() fragment_instance_id=54366bcc3d7a4915-a3cfb03428ce250f
I0215 17:37:10.908537 49070 query_fragments_ctx.h:55] Deregister query/load memory tracker, queryId=54366bcc3d7a4915-a3cfb03428ce250e, Limit=2.00 GB, CurrUsed=851.44 KB, PeakUsed=1.11 MB
I0215 17:37:14.974335 48861 daemon.cpp:217] physical memory 15.65 GB, process memory used 1.74 GB limit 12.52 GB, sys mem available 11.47 GB low water mark 1.57 GB, refresh interval memory growth 0 B
I0215 17:37:15.423421 49444 task_worker_pool.cpp:1519] successfully report TASK|host=192.168.129.101|port=9020
I0215 17:37:19.989881 48861 daemon.cpp:217] physical memory 15.65 GB, process memory used 1.74 GB limit 12.52 GB, sys mem available 11.47 GB low water mark 1.57 GB, refresh interval memory growth 0 B
I0215 17:37:22.674818 49364 olap_server.cpp:719] cooldown producer get tablet num: 0
I0215 17:37:25.005756 48861 daemon.cpp:217] physical memory 15.65 GB, process memory used 1.74 GB limit 12.52 GB, sys mem available 11.47 GB low water mark 1.57 GB, refresh interval memory growth 0 B

**Comment by 442496146** [comment]
200万行都是秒成功。3-400万行数据量就会报错，select从catalog mysql外表insert到doris表

**Comment by zixuedanxin** [comment]
查询MYSQL的数据库报OutOfMemoryError的错误

为减少内存的使用，在获取结果集时，每次仅获取batchSize的大小，这样一批一批的获取结果。而MYSQL默认是一次将结果全部加载到内存， 设置的按批获取无法生效，需要主动显示的在URL中指定:"jdbc_url"="jdbc:mysql://IP:PORT/doris_test?useCursorFetch=true"

**Reply by 442496146** [reply]
无效，我改了be的xmx也不行，剩余内存挺多的

**Comment by 442496146** [comment, marked as answer]
1.2.1版本有，而且cpu飙高。1.2.2版本均未出现
